framework/src/mesh/FileMesh.C

```diff
@@ -74,6 +82,30 @@ FileMesh::buildMesh()
     getMesh().prepare_for_use();
     getMesh().skip_partitioning(skip_partitioning_later);
   }
+  else if (_is_checkpoint)
+  {
+    CheckpointIO reader(getMesh());
+
+    reader.binary() = true;
```

Comment:
Could this be determined via file name extension when reading, and via some API when writing?  ASCII files end up being just as small when gzip is used, and are much easier to debug with when something goes wrong.  Since CheckpointIO is explicitly *not* supporting backwards compatibility when we change formats, and since I expect users to be unaware of or to ignore that, "something goes wrong" may be a common case.